compiler/ballerina-lang/src/main/java/io/ballerina/projects/CodeGeneratorManager.java

```diff
@@ -267,6 +275,15 @@ public PackageCompilation compilation() {
 
         @Override
         public void addSourceFile(TextDocument textDocument, String filenamePrefix, ModuleId moduleId) {
+            if (this.currentPackage.project().kind().equals(ProjectKind.SINGLE_FILE_PROJECT)) {
+                DiagnosticInfo diagnosticInfo = new DiagnosticInfo(
+                        ProjectDiagnosticErrorCode.UNSUPPORTED_COMPILER_PLUGIN_TYPE.diagnosticId(),
+                        "skipping addition of generated file '" + filenamePrefix +
```

Comment:
How about the following error message? 

"Skipped adding the generated source file with prefix "dummyfunc". Source file generation is not supported with standalone bal files"